napari/layers/utils/style_encoding.py

```diff
@@ -100,7 +100,15 @@ def _json_encode(self) -> dict:
         """
 
 
-class _ConstantStyleEncoding(EventedModel, Generic[StyleValue, StyleArray]):
+class _StyleEncodingModel(EventedModel):
+    class Config:
+        # Match fields exactly so that dicts can be sensibly parsed.
+        extra = 'forbid'
+
+
+class _ConstantStyleEncoding(
+    _StyleEncodingModel, Generic[StyleValue, StyleArray]
```

Comment:
Can you also briefly explain in a comment above how Generic works? Maybe with a link to the Python typing docs? @tlambert03 has sometimes been frustrated when I ask that comments explain "standard" Python features, but my feeling is that there are standard but *uncommon* features that should be explained when they are used. If someone with my Python experience hasn't encountered them before, then that's a pretty good indicator that others will also be confused when reading the code, and a clarifying comment is warranted.

Reply:
Sure. In my description, I'm slightly questioning the specific types these take on `StringEncoding` and `ColorEncoding`. But ultimately I'm trying to get across that calling these types of encodings will return array likes with certain dtypes and shapes.

Reply:
Done.